src/InstrumentationEngine.Attach/Program.cs

```diff
@@ -316,11 +316,25 @@ private static void WriteError(string message)
                                 {
                                     foreach (var setting in source.Settings)
                                     {
-                                        methodSettings.Add(new SettingsTypeSetting()
+                                        if (setting.IsPathSpecified &&
```

Comment:
nit: Maybe have the if-else assign to a "string settingValue" local and then only write the "methodsSettings.Add(...)" code once.